### ISSUE TITLE
[no-extra-parens] Incorrect rule flagging when using '??'

### ISSUE THREAD

**Issue by zajrik** [typescript-eslint/typescript-eslint#1155]
**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/no-extra-parens": ["warn"]
  }
}
```

```TS
public constructor(dir: string, keepDays?: number, level?: LogLevel)
{
    super(level);
    this._logDir = dir;

    // eslint-disable-next-line @typescript-eslint/no-extra-parens
    this._keepDays = Math.max(0, (keepDays ?? 7) - 1);
    // We receive a warning here ^

    if (!FS.existsSync(this._logDir))
        FS.mkdirSync(this._logDir);

    this._newLog();
}
```

**Expected Result**
I expect that it should not give a warning, as `Math.max(0, (keepDays || 7) - 1)` is allowed.

**Actual Result**
A warning is given. The parens are necessary to produce the desired result because without them it compiles to `Math.max(0, (keepDays !== null && keepDays !== void 0 ? keepDays : 7 - 1))` which produces undesired behavior due to the subtraction taking precedence.

**Additional Info**
Now that `??` is available I want to use it in cases where `||` would normally be used for null-coalescence, as I prefer not to use `||` outside of actual boolean operations because I don't like using truthy/falsey values in favor of doing actual strict null/undefined checks.

Edit: I've realized I can work around this by refactoring to `Math.max(1, keepDays ?? 7) - 1` which produces the same desired result of clamping the number to a positive integer greater than or equal to 0. Obviously the issue is still prevalent in cases where one couldn't refactor to keep the desired result while still keeping the simplicity that nullish-coalescence affords.

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.5.0` |
| `@typescript-eslint/parser`        | `2.5.0` |
| `TypeScript`                       | `3.7.1-rc` |
| `ESLint`                           | `6.6.0` |
| `node`                             | `12.13.0, 10.16.0` |
| `npm`                              | `6.12.0, 6.9.0` |

**Comment by bradzacher** [comment]
This is a problem that can be fixed in the base eslint framework:
https://github.com/eslint/eslint/blob/0a480f8307a0e438032f484254941e6426748143/lib/rules/utils/ast-utils.js#L791-L827

They don't yet have support for `??` binary expression precedence

**Comment by yeonjuan** [comment]
🚀 [ESLint 7.2.0](https://github.com/eslint/eslint/releases/tag/v7.2.0) is released and it support `??`.

**Comment by bradzacher** [comment]
Yup! It looks like this is fixed in the base rule.
[repl](https://eslint.org/demo#eyJ0ZXh0IjoiLyogZXNsaW50IG5vLWV4dHJhLXBhcmVuczogWydlcnJvciddICovXG5mdW5jdGlvbiBmb28oZGlyLCBrZWVwRGF5cywgbGV2ZWwpXG57XG4gICAgdGhpcy5fbG9nRGlyID0gZGlyO1xuXG4gICAgdGhpcy5fa2VlcERheXMgPSBNYXRoLm1heCgwLCAoa2VlcERheXMgPz8gNykgLSAxKTtcbiAgICAvLyBXZSByZWNlaXZlIGEgd2FybmluZyBoZXJlIF5cblxuICAgIGlmICghRlMuZXhpc3RzU3luYyh0aGlzLl9sb2dEaXIpKVxuICAgICAgICBGUy5ta2RpclN5bmModGhpcy5fbG9nRGlyKTtcblxuICAgIHRoaXMuX25ld0xvZygpO1xufSIsIm9wdGlvbnMiOnsicGFyc2VyT3B0aW9ucyI6eyJlY21hVmVyc2lvbiI6MTEsInNvdXJjZVR5cGUiOiJzY3JpcHQiLCJlY21hRmVhdHVyZXMiOnt9fSwicnVsZXMiOnt9LCJlbnYiOnt9fX0=)

That's great to hear that they've supported it now :)
